### PR TITLE
SwitchCaseSemicolonToColonFixer - Add

### PR DESCRIPTION
**Pull request by SpacePossum**
@ see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1658

(no worries, this will be my last PR until some the open ones are done ^-^)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -284,6 +284,10 @@ Choose from the list of available fixers:
                         one blank line after the use
                         statements block.
 
+* **switch_case_colon** [PSR-2]
```
Comment (keradus) [inline README.rst]: name does not says what would be "fixed" version

Comment (SpacePossum) [inline README.rst]: would renaming `switch_case_colon` to `switch_case_semicolon_to_colon` be better?

Comment (keradus) [inline README.rst]: :+1:

Comment (SpacePossum) [inline README.rst]: :+1: done

File: Symfony/CS/Fixer/PSR2/SwitchCaseSemicolonToColonFixer.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\PSR2;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
```
Comment (keradus) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSemicolonToColonFixer.php]: For PSR2-level fixers we put a reference to paragraph from standard in the docblock

**Comment by keradus** [comment]
a minimal case with nested switches would be great

**Comment by SpacePossum** [comment]
also needs an integration test, this fixer should run after `DuplicateSemicolonFixer`

File: Symfony/CS/Fixer/PSR2/SwitchCaseSemicolonToColonFixer.php (hunk 2 above)
Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSemicolonToColonFixer.php]: :+1:  done, I also added some other ones that were missing.

**Comment by SpacePossum** [comment]
>a minimal case with nested switches would be great

:+1:  Done :)

File: Symfony/CS/Tests/Fixer/PSR2/SwitchCaseSpaceFixerTest.php [hunk 3]
```diff
@@ -34,14 +34,6 @@ public function provideFixCases()
             array(
                 '<?php
                 switch ($a) {
-                    case "prod":
```
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/PSR2/SwitchCaseSpaceFixerTest.php]: duplicate test case

File: Symfony/CS/Tests/Fixtures/Integration/priority/duplicate_semicolon,switch_case_semicolon_to_colon.test [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+switch($a) {
+    case 1:
+        echo 2;
+}
+--INPUT--
+<?php
+switch($a) {
+    case 1;;
+        echo 2;
+}
```
Comment (keradus) [inline Symfony/CS/Tests/Fixtures/Integration/priority/duplicate_semicolon,switch_case_semicolon_to_colon.test]: please add new line at the end of file

**Comment by keradus** [comment]
:+1: (after that one minor comment)

File: Symfony/CS/Tests/Fixtures/Integration/priority/duplicate_semicolon,switch_case_semicolon_to_colon.test (hunk 4 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixtures/Integration/priority/duplicate_semicolon,switch_case_semicolon_to_colon.test]: :+1:  please see https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/3f5bbb3c0eb4643b7a6eb4787aea1693fe166b1d

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**